squash-honeycomb.ipynb

```diff
@@ -0,0 +1,381 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "SQUASH_API_URL = \"https://squash-restful-api-demo.lsst.codes/\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "r = requests.get(SQUASH_API_URL + \"/job/1\").json()\n",
+    "    \n",
+    "    \n",
+    "data = {'id': r['id'],\n",
+    "        'date_created': r['date_created'],\n",
+    "        'filter_name': r['meta']['filter_name'],\n",
+    "        'dataset': r['ci_dataset']}\n",
+    "            \n",
+    "# events should be flat dict\n",
+    "for meas in r['measurements']:\n",
+    "    data[meas['metric']] = meas['value']\n",
+    "    \n",
+    "data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This looks like an structured log entry, it also looks like a property set or it may recall something else..."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Sending events to Honeycomb\n",
+    "\n",
+    "1. When honeycomb gets a request to add an event to a dataset that doesn't exist yet, it creates the dataset.\n",
+    "2. The dataset schema is inferred automatically. It also will infer automatically if you add new fields to your events.\n",
+    "\n",
+    "3. Events are added to the dataset for querying.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "NOTE: In order to execute this cell you will need the HONEY_API_KEY. It is avaibale from \"Team Settings\" in the UI once you join the `lsst-square` team."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import libhoney\n",
+    "\n",
+    "HONEY_API_KEY = \"\"  # Obtained from the UI under \"Team Settings\"\n",
+    "\n",
+    "libhoney.init(writekey=HONEY_API_KEY, dataset=\"single-event-demo\")\n",
+    "libhoney.send_now(data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "That's it! Check the new dataset at https://ui.honeycomb.io/lsst-square/datasets"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "NOTE: this demo dataset may already exist from previous executions of this notebook. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Now, let's send all squash data to Honeycomb\n",
+    "Actually, just the scalar metrics and some context information."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Create the `squash-demo` dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "libhoney.init(writekey=HONEY_API_KEY, dataset=\"squash-demo\")\n",
```

Comment:
Hmm.  I wanted to create my own dataset, so I changed squash-demo to squash-demo-simon.  It generated the dataset, but none of the events in the next cell showed up in the dataset.

Reply:
OK, this is the default range being 2 hrs ago.  It would probably be good to either adjust the timestamps in the events or explain that the query range must be altered.